raiden/smart_contracts/ChannelManagerLibrary.sol

```diff
@@ -7,36 +7,59 @@ library ChannelManagerLibrary {
     // TODO: experiment with a sorted data structure
     struct Data {
         mapping(address => address[]) node_channels;
+        mapping(bytes32 => address) channel_addresses;
         address[] all_channels;
         Token token;
     }
 
     /// @notice getChannelsAddresses to get all channels
     /// @dev Get all channels
-    /// @return channels (address[]) all the channels
-    function getChannelsAddresses(Data storage self) returns (address[] channels) {
+    /// @return all the open channels
+    function getChannelsAddresses(Data storage self) constant returns (address[] channels) {
```

Comment:
I believe this function should be removed, and the ChannelManagerContract can expose this data directly